src/sonic-config-engine/sonic_device_util.py

```diff
@@ -29,6 +31,39 @@ def get_machine_info():
             machine_vars[tokens[0]] = tokens[1].strip()
     return machine_vars
 
+def get_num_npus():
+   platform = get_platform_info(get_machine_info)
+   num_npus = 1
+   asic_conf_file_path = os.path.join(SONIC_DEVICE_PATH, platform, 'asic.conf')
+   if not os.path.isFile(asic_conf_file_path):
+	return num_npus
+   with open(asic_conf_file_path) as asic_conf_file:
+	for line in asic_conf_file:
```

Comment:
I'd suggest to use some format here, and don't parse it manually. 
json or yaml will work

Reply:
The file 'asic.conf' is being parsed in systemd-generator and other places as well.
I will change the format of this file and corresponding changes in the places it is being used in an  future PR.